modules/transcription-service-persistence/src/main/java/org/opencastproject/transcription/persistence/TranscriptionJobControlDto.java

```diff
@@ -0,0 +1,246 @@
+          Date dateCompleted, String status, long trackDuration, long providerId) {
+    super();
+    this.mediaPackageId = mediaPackageId;
+    this.trackId = trackId;
+    this.transcriptionJobId = transcriptionJobId;
+    this.dateCreated = dateCreated;
+    this.dateCompleted = dateCompleted;
+    this.status = status;
+    this.trackDuration = trackDuration;
+    this.providerId = providerId;
+  }
+
+  /**
+   * Convert into business object.
+   */
+  public TranscriptionJobControl toTranscriptionJobControl() {
+    return new TranscriptionJobControl(mediaPackageId, trackId, transcriptionJobId, dateCreated, dateCompleted, status,
+            trackDuration, providerId);
+  }
+
+  /**
+   * Store new job control
+   */
+  public static TranscriptionJobControlDto store(EntityManager em, String mediaPackageId, String trackId,
+          String transcriptionJobId, String jobStatus, long trackDuration, long providerId) throws TranscriptionDatabaseException {
+    TranscriptionJobControlDto dto = new TranscriptionJobControlDto(mediaPackageId, trackId, transcriptionJobId,
+            new Date(), null, jobStatus, trackDuration, providerId);
+
+    EntityTransaction tx = em.getTransaction();
+    try {
+      tx.begin();
+      em.persist(dto);
+      tx.commit();
+      return dto;
+    } catch (Exception e) {
+      if (tx.isActive()) {
+        tx.rollback();
+      }
+      throw new TranscriptionDatabaseException(e);
+    } finally {
+      em.close();
+    }
+  }
+
+  /**
+   * Find a job control by its number.
+   */
+  public static TranscriptionJobControlDto findByJob(EntityManager em, String jobId)
+          throws TranscriptionDatabaseException {
+    Query query = null;
+    try {
+      query = em.createNamedQuery("TranscriptionJobControl.findByJob");
+      query.setParameter("transcriptionJobId", jobId);
+      return (TranscriptionJobControlDto) query.getSingleResult();
+    } catch (NoResultException e) {
+      return null; // Not found
+    } catch (Exception e) {
+      throw new TranscriptionDatabaseException(e);
+    }
+  }
+
+  /**
+   * Find all job controls by media package.
+   */
+  @SuppressWarnings("unchecked")
+  public static List<TranscriptionJobControlDto> findByMediaPackage(EntityManager em, final String mediaPackageId)
+          throws TranscriptionDatabaseException {
+    Query query = null;
+    try {
+      query = em.createNamedQuery("TranscriptionJobControl.findByMediaPackage");
+      query.setParameter("mediaPackageId", mediaPackageId);
+      return query.getResultList();
+    } catch (Exception e) {
+      throw new TranscriptionDatabaseException(e);
+    }
+  }
+
+  /**
+   * Find all job controls by status.
+   */
+  @SuppressWarnings("unchecked")
+  public static List<TranscriptionJobControlDto> findByStatus(EntityManager em, final String... status)
+          throws TranscriptionDatabaseException {
+    Collection<String> statusCol = new HashSet<String>();
+    for (String st : status) {
```

Comment:
I see what you're doing, and understand why.  I would still break this into two methods, one which takes a single Status, the other which takes a Set<Status>.  The single input would wrap its input into a Set, then call the Set input to keep the logic in one place.

Hint: What if status is `null`?

Reply:
I don't understand the 2nd sentence. 
The current implementation will handle both single status and a set of statuses.  I've added a check for null status. 
What's the reasoning behind splitting into 2 methods?

Reply:
Right now, if I were to call this method with a null as the final parameter, I would see a null pointer exception on 176 when Java attempts to iterate on a null.  I don't see a null check, but that might be a local change that's not here yet.  If it's checked we can leave it as is.

Reply:
changed was local, I've now pushed it.